### DISCUSSION TITLE
Autonomous flight using the ROS packages and Flow Deck

### DISCUSSION THREAD

**Discussion by santiagorg2401** [USC-ACTLab/crazyswarm#314] [Q&A]
Hi,
I followed the steps in https://crazyswarm.readthedocs.io/en/latest/usage.html, but I have a problem, by the moment I am using Flow Deck but the goal is to use LPS, so after tagging the Crazyflies with 1,2,3,4,..., A, B, and writing them in allCrazyflies.yaml
crazyflies:1
...
crazyflies : 10
...
and so on
The GUI chooser.py shows me this error when clicking any button:
[rospack] Error: package 'crazyflie_tools' not found
Exception in Tkinter callback
Traceback (most recent call last):
File "/usr/lib/python2.7/lib-tk/Tkinter.py", line 1550, in __call__
return self.func(*args)
File "chooser.py", line 213, in checkBattery
voltage = subprocess.check_output(["rosrun crazyflie_tools battery --uri " + uri], shell=True)
File "/usr/lib/python2.7/subprocess.py", line 223, in check_output
raise CalledProcessError(retcode, cmd, output=output)
CalledProcessError: Command '['rosrun crazyflie_tools battery --uri radio://0/80/2M/E7E7E7E708']' returned non-zero exit status 2

Then, when I try to fly the drone with (for example) niceHover.py, first I run the crayswarm server
auto-starting new master
process[master]: started with pid [9403]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to d6b7ebd0-c611-11ea-bc79-181deab367da
process[rosout-1]: started with pid [9414]
started core service [/rosout]
process[crazyswarm_server-2]: started with pid [9421]
ERROR: cannot launch node of type [joy/joy_node]: joy
ROS path [0]=/opt/ros/melodic/share/ros
ROS path [1]=/home/santiagorg2401/crazyflie/crazyswarm/ros_ws/src
ROS path [2]=/opt/ros/melodic/share
process[crazyswarm_teleop-4]: started with pid [9422]
process[rviz-5]: started with pid [9423]
[ INFO] [1594759051.849131804]: Wait for services...
[ INFO] [1594759051.851109082]: waitForService: Service [/emergency] has not been advertised, waiting...
[ INFO] [1594759051.918606844]: waitForService: Service [/emergency] is now available.
ch: 1
[ INFO] [1594759051.932975640]: Manager ready.
terminate called after throwing an instance of 'XmlRpc::XmlRpcException'
[crazyswarm_server-2] process has died [pid 9421, exit code -6, cmd /home/santiagorg2401/crazyflie/crazyswarm/ros_ws/devel/lib/crazyswarm/crazyswarm_server __name:=crazyswarm_server __log:=/home/santiagorg2401/.ros/log/d6b7ebd0-c611-11ea-bc79-181deab367da/crazyswarm_server-2.log].
log file: /home/santiagorg2401/.ros/log/d6b7ebd0-c611-11ea-bc79-181deab367da/crazyswarm_server-2*.log

And finally in another terminal window, the niceHover.py
Traceback (most recent call last):
File "niceHover.py", line 9, in <module>
swarm = Crazyswarm()
File "/home/santiagorg2401/crazyflie/crazyswarm/ros_ws/src/crazyswarm/scripts/pycrazyswarm/crazyswarm.py", line 43, in __init__
import crazyflie
File "/home/santiagorg2401/crazyflie/crazyswarm/ros_ws/src/crazyswarm/scripts/pycrazyswarm/crazyflie.py", line 12, in <module>
from crazyflie_driver.srv import *
ImportError: No module named crazyflie_driver.srv

I do not really know why is those errors happening, please let me know if I am forgetting to do/execute something or if you have any idea to help.
Thank you all, have a nice day

**Comment by whoenig** [comment]
For your first problem, make sure you source your ROS workspace (`source ~/crazyswarm/ros_ws/source/devel.sh`) in every terminal that you use. For your second problem, there seems to be an issue with the formatting of your configuration files (either `hover_swarm.launch` or `crazyflies.yaml`) -- please post/attach the files here.

**Comment by santiagorg2401** [comment]
crazyflies.yaml
crazyflies:
- channel: 80
  id: 8
  initialPosition: [0, 0, 0.0]
  type: default

-hover_swarm.launch

<?xml version="1.0"?>
<launch>
  <arg name="joy_dev" default="/dev/input/js0" />

  <rosparam command="load" file="$(find crazyswarm)/launch/crazyflieTypes.yaml" />
  <rosparam command="load" file="$(find crazyswarm)/launch/crazyflies.yaml" />

  <node pkg="crazyswarm" type="crazyswarm_server" name="crazyswarm_server" output="screen" >
    <rosparam>
      world_frame: "/world"
      # Logging configuration (Use enable_logging to actually enable logging)
      genericLogTopics: ["log1"]
      genericLogTopicFrequencies: [10]
      genericLogTopic_log1_Variables: ["stateEstimate.x", "ctrltarget.x"]
      # firmware parameters for all drones (use crazyflieTypes.yaml to set per type, or
      # allCrazyflies.yaml to set per drone)
      firmwareParams:
        commander:
          enHighLevel: 1
        stabilizer:
          estimator: 2 # 1: complementary, 2: kalman
          controller: 2 # 1: PID, 2: mellinger
        ring:
          effect: 16 # 6: double spinner, 7: solid color, 16: packetRate
          solidBlue: 255 # if set to solid color
          solidGreen: 0 # if set to solid color
          solidRed: 0 # if set to solid color
          headlightEnable: 0
        locSrv:
          extPosStdDev: 1e-3
          extQuatStdDev: 0.5e-1
        kalman:
          resetEstimation: 1
      # tracking
      motion_capture_type: "none" # one of none,vicon,optitrack,qualisys,vrpn
      object_tracking_type: "libobjecttracker" # one of motionCapture,libobjecttracker
      send_position_only: False # set to False to send position+orientation; set to True to send position only
      vicon_host_name: "vicon"
      # optitrack_host_name: "optitrack"
      # qualisys_host_name: "10.0.5.219"
      # qualisys_base_port: 22222
      # vrpn_host_name: "vicon"
      save_point_clouds: ~/pointCloud.ot
      print_latency: False
      write_csvs: False
      force_no_cache: False
      enable_parameters: True
      enable_logging: False
    </rosparam>
  </node>

  <node name="joy" pkg="joy" type="joy_node" output="screen">
    <param name="dev" value="$(arg joy_dev)" />
  </node>

  <node pkg="crazyswarm" type="crazyswarm_teleop" name="crazyswarm_teleop" output="screen">
    <param name="csv_file" value="$(find crazyswarm)/launch/figure8_smooth.csv" />
    <param name="timescale" value="0.8" />
  </node>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find crazyswarm)/launch/test.rviz"/>

  
  

</launch>

**Comment by whoenig** [comment]
Please use "insert code" in the toolbar, or upload the files somewhere and share the link. The configuration is whitespace sensitive, so I can't tell if there is an issue from your previous comment.

**Comment by santiagorg2401** [comment]
[Files.zip](https://github.com/USC-ACTLab/crazyswarm/files/4962874/Files.zip)

**Comment by whoenig** [comment]
The only thing I spot is the initialPosition in `crazyflies.yaml`. I remember there being a parsing bug in ROS that requires this to be a floating point number, i.e., `[0.0, 0.0, 0.0]` rather than `[0, 0, 0.0]`.

**Comment by santiagorg2401** [comment]
![image](https://user-images.githubusercontent.com/52050852/88403120-b2af3400-cd91-11ea-8a18-5b5b750df8e9.png)
Terminal 1.
`santiagorg2401@santiagorg2401-G7-7588:~$ source crazyflie/crazyswarm/ros_ws/devel/setup.bash 
santiagorg2401@santiagorg2401-G7-7588:~$ python crazyflie/crazyswarm/ros_ws/src/crazyswarm/scripts/chooser.py 
timeout
Exception in Tkinter callback
Traceback (most recent call last):
  File "/usr/lib/python2.7/lib-tk/Tkinter.py", line 1550, in __call__
    return self.func(*args)
  File "crazyflie/crazyswarm/ros_ws/src/crazyswarm/scripts/chooser.py", line 213, in checkBattery
    voltage = subprocess.check_output(["rosrun crazyflie_tools battery --uri " + uri], shell=True)
  File "/usr/lib/python2.7/subprocess.py", line 223, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
CalledProcessError: Command '['rosrun crazyflie_tools battery --uri radio://0/80/2M/E7E7E7E706']' returned non-zero exit status 1
6
timeout
timeout
Exception in Tkinter callback
Traceback (most recent call last):
  File "/usr/lib/python2.7/lib-tk/Tkinter.py", line 1550, in __call__
    return self.func(*args)
  File "crazyflie/crazyswarm/ros_ws/src/crazyswarm/scripts/chooser.py", line 213, in checkBattery
    voltage = subprocess.check_output(["rosrun crazyflie_tools battery --uri " + uri], shell=True)
  File "/usr/lib/python2.7/subprocess.py", line 223, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
CalledProcessError: Command '['rosrun crazyflie_tools battery --uri radio://0/80/2M/E7E7E7E706']' returned non-zero exit status 1
Flash STM32 FW to radio://0/80/2M/E7E7E7E706
timeout
santiagorg2401@santiagorg2401-G7-7588:~$ python crazyflie/crazyswarm/ros_ws/src/crazyswarm/scripts/chooser.py 
`
Terminal 2.
`santiagorg2401@santiagorg2401-G7-7588:~$ source crazyflie/crazyswarm/ros_ws/devel/setup.bash 
santiagorg2401@santiagorg2401-G7-7588:~$ roslaunch crazyswarm hover_swarm.launch 
... logging to /home/santiagorg2401/.ros/log/3456f9ce-cdbb-11ea-ab55-181deab367da/roslaunch-santiagorg2401-G7-7588-5535.log
Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server http://santiagorg2401-G7-7588:46631/

SUMMARY
========

PARAMETERS
 * /crazyflieTypes/CF21SingleMarker/batteryVoltageWarning: 3.8
 * /crazyflieTypes/CF21SingleMarker/batteryVoltateCritical: 3.7
 * /crazyflieTypes/CF21SingleMarker/bigQuad: False
 * /crazyflieTypes/CF21SingleMarker/dynamicsConfiguration: 0
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/i_range_m_z: 1500
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/i_range_xy: 2.0
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/i_range_z: 0.4
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kR_xy: 70000
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kR_z: 60000
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kd_omega_rp: 200
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kd_xy: 0.2
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kd_z: 0.4
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/ki_m_z: 500
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/ki_xy: 0.05
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/ki_z: 0.05
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kp_xy: 0.4
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kp_z: 1.25
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kw_xy: 20000
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kw_z: 12000
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/mass: 0.032
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/massThrust: 132000
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/kalman/mNGyro_rollpitch: 0.1
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/kalman/mNGyro_yaw: 0.1
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/kalman/pNAcc_xy: 1.0
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/kalman/pNAcc_z: 2.0
 * /crazyflieTypes/CF21SingleMarker/markerConfiguration: 3
 * /crazyflieTypes/default/batteryVoltageWarning: 3.8
 * /crazyflieTypes/default/batteryVoltateCritical: 3.7
 * /crazyflieTypes/default/bigQuad: False
 * /crazyflieTypes/default/dynamicsConfiguration: 0
 * /crazyflieTypes/default/firmwareParams/ctrlMel/i_range_m_z: 1500
 * /crazyflieTypes/default/firmwareParams/ctrlMel/i_range_xy: 2.0
 * /crazyflieTypes/default/firmwareParams/ctrlMel/i_range_z: 0.4
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kR_xy: 70000
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kR_z: 60000
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kd_omega_rp: 200
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kd_xy: 0.2
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kd_z: 0.4
 * /crazyflieTypes/default/firmwareParams/ctrlMel/ki_m_z: 500
 * /crazyflieTypes/default/firmwareParams/ctrlMel/ki_xy: 0.05
 * /crazyflieTypes/default/firmwareParams/ctrlMel/ki_z: 0.05
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kp_xy: 0.4
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kp_z: 1.25
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kw_xy: 20000
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kw_z: 12000
 * /crazyflieTypes/default/firmwareParams/ctrlMel/mass: 0.032
 * /crazyflieTypes/default/firmwareParams/ctrlMel/massThrust: 132000
 * /crazyflieTypes/default/markerConfiguration: 0
 * /crazyflieTypes/defaultSingleMarker/batteryVoltageWarning: 3.8
 * /crazyflieTypes/defaultSingleMarker/batteryVoltateCritical: 3.7
 * /crazyflieTypes/defaultSingleMarker/bigQuad: False
 * /crazyflieTypes/defaultSingleMarker/dynamicsConfiguration: 0
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/i_range_m_z: 1500
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/i_range_xy: 2.0
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/i_range_z: 0.4
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kR_xy: 70000
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kR_z: 60000
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kd_omega_rp: 200
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kd_xy: 0.2
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kd_z: 0.4
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/ki_m_z: 500
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/ki_xy: 0.05
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/ki_z: 0.05
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kp_xy: 0.4
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kp_z: 1.25
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kw_xy: 20000
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kw_z: 12000
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/mass: 0.032
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/massThrust: 132000
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/kalman/mNGyro_rollpitch: 0.2
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/kalman/mNGyro_yaw: 0.1
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/kalman/pNAcc_xy: 4.0
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/kalman/pNAcc_z: 8.0
 * /crazyflieTypes/defaultSingleMarker/markerConfiguration: 3
 * /crazyflieTypes/large/batteryVoltageWarning: 11.4
 * /crazyflieTypes/large/batteryVoltateCritical: 11.1
 * /crazyflieTypes/large/bigQuad: True
 * /crazyflieTypes/large/dynamicsConfiguration: 0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/i_range_m_z: 1500
 * /crazyflieTypes/large/firmwareParams/ctrlMel/i_range_xy: 2.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/i_range_z: 4.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kR_xy: 30000
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kR_z: 40000
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kd_omega_rp: 100
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kd_xy: 3.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kd_z: 3.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/ki_m_z: 500
 * /crazyflieTypes/large/firmwareParams/ctrlMel/ki_xy: 1.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/ki_z: 2.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kp_xy: 8.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kp_z: 8.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kw_xy: 9000
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kw_z: 10000
 * /crazyflieTypes/large/firmwareParams/ctrlMel/mass: 0.38
 * /crazyflieTypes/large/firmwareParams/ctrlMel/massThrust: 5400
 * /crazyflieTypes/large/markerConfiguration: 2
 * /crazyflieTypes/medium/batteryVoltageWarning: 7.6
 * /crazyflieTypes/medium/batteryVoltateCritical: 7.4
 * /crazyflieTypes/medium/bigQuad: True
 * /crazyflieTypes/medium/dynamicsConfiguration: 0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/i_range_m_z: 1500
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/i_range_xy: 2.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/i_range_z: 2.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kR_xy: 50000
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kR_z: 60000
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kd_omega_rp: 200
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kd_xy: 1.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kd_z: 1.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/ki_m_z: 500
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/ki_xy: 0.1
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/ki_z: 0.5
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kp_xy: 2.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kp_z: 3
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kw_xy: 16000
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kw_z: 12000
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/mass: 0.11
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/massThrust: 23000
 * /crazyflieTypes/medium/markerConfiguration: 1
 * /crazyflies: [{'initialPositio...
 * /crazyswarm_server/enable_logging: False
 * /crazyswarm_server/enable_parameters: True
 * /crazyswarm_server/firmwareParams/commander/enHighLevel: 1
 * /crazyswarm_server/firmwareParams/kalman/resetEstimation: 1
 * /crazyswarm_server/firmwareParams/locSrv/extPosStdDev: 1e-3
 * /crazyswarm_server/firmwareParams/locSrv/extQuatStdDev: 0.05
 * /crazyswarm_server/firmwareParams/ring/effect: 16
 * /crazyswarm_server/firmwareParams/ring/headlightEnable: 0
 * /crazyswarm_server/firmwareParams/ring/solidBlue: 255
 * /crazyswarm_server/firmwareParams/ring/solidGreen: 0
 * /crazyswarm_server/firmwareParams/ring/solidRed: 0
 * /crazyswarm_server/firmwareParams/stabilizer/controller: 2
 * /crazyswarm_server/firmwareParams/stabilizer/estimator: 2
 * /crazyswarm_server/force_no_cache: False
 * /crazyswarm_server/genericLogTopicFrequencies: [10]
 * /crazyswarm_server/genericLogTopic_log1_Variables: ['stateEstimate.x...
 * /crazyswarm_server/genericLogTopics: ['log1']
 * /crazyswarm_server/motion_capture_type: none
 * /crazyswarm_server/object_tracking_type: libobjecttracker
 * /crazyswarm_server/print_latency: False
 * /crazyswarm_server/save_point_clouds: ~/pointCloud.ot
 * /crazyswarm_server/send_position_only: False
 * /crazyswarm_server/vicon_host_name: vicon
 * /crazyswarm_server/world_frame: /world
 * /crazyswarm_server/write_csvs: False
 * /crazyswarm_teleop/csv_file: /home/santiagorg2...
 * /crazyswarm_teleop/timescale: 0.8
 * /dynamicsConfigurations/0/maxFitnessScore: 0.001
 * /dynamicsConfigurations/0/maxPitch: 1.4
 * /dynamicsConfigurations/0/maxPitchRate: 20.0
 * /dynamicsConfigurations/0/maxRoll: 1.4
 * /dynamicsConfigurations/0/maxRollRate: 20.0
 * /dynamicsConfigurations/0/maxXVelocity: 2.0
 * /dynamicsConfigurations/0/maxYVelocity: 2.0
 * /dynamicsConfigurations/0/maxYawRate: 10.0
 * /dynamicsConfigurations/0/maxZVelocity: 3.0
 * /joy/dev: /dev/input/js0
 * /markerConfigurations/0/numPoints: 4
 * /markerConfigurations/0/offset: [0.0, -0.01, -0.04]
 * /markerConfigurations/0/points/0: [0.0177184, 0.013...
 * /markerConfigurations/0/points/1: [-0.0262914, 0.05...
 * /markerConfigurations/0/points/2: [-0.0328889, -0.0...
 * /markerConfigurations/0/points/3: [0.0431307, -0.03...
 * /markerConfigurations/1/numPoints: 4
 * /markerConfigurations/1/offset: [0.0, 0.0, -0.03]
 * /markerConfigurations/1/points/0: [-0.00896228, -0....
 * /markerConfigurations/1/points/1: [-0.0156318, 0.09...
 * /markerConfigurations/1/points/2: [0.0461693, -0.08...
 * /markerConfigurations/1/points/3: [-0.0789959, -0.0...
 * /markerConfigurations/2/numPoints: 4
 * /markerConfigurations/2/offset: [0.0, 0.0, -0.06]
 * /markerConfigurations/2/points/0: [0.0558163, -0.00...
 * /markerConfigurations/2/points/1: [-0.0113941, 0.00...
 * /markerConfigurations/2/points/2: [-0.0306277, 0.05...
 * /markerConfigurations/2/points/3: [0.0535816, -0.04...
 * /markerConfigurations/3/numPoints: 1
 * /markerConfigurations/3/offset: [0.0, -0.01, -0.04]
 * /markerConfigurations/3/points/0: [0.0177184, 0.013...
 * /numDynamicsConfigurations: 1
 * /numMarkerConfigurations: 4
 * /rosdistro: melodic
 * /rosversion: 1.14.6

NODES
  /
    crazyswarm_server (crazyswarm/crazyswarm_server)
    crazyswarm_teleop (crazyswarm/crazyswarm_teleop)
    joy (joy/joy_node)
    rviz (rviz/rviz)

auto-starting new master
process[master]: started with pid [5547]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to 3456f9ce-cdbb-11ea-ab55-181deab367da
process[rosout-1]: started with pid [5558]
started core service [/rosout]
process[crazyswarm_server-2]: started with pid [5565]
ERROR: cannot launch node of type [joy/joy_node]: joy
ROS path [0]=/opt/ros/melodic/share/ros
ROS path [1]=/home/santiagorg2401/crazyflie/crazyswarm/ros_ws/src
ROS path [2]=/opt/ros/melodic/share
process[crazyswarm_teleop-4]: started with pid [5566]
process[rviz-5]: started with pid [5568]
[ INFO] [1595601452.053808269]: Wait for services...
[ INFO] [1595601452.055580089]: waitForService: Service [/emergency] has not been advertised, waiting...
[ INFO] [1595601452.206348997]: waitForService: Service [/emergency] is now available.
[ INFO] [1595601452.221361957]: Manager ready.
ch: 1
[ INFO] [1595601452.235945328]: Adding CF: cf6 (radio://0/80/2M/E7E7E7E706, cf6)...
terminate called after throwing an instance of 'std::runtime_error'
  what():  timeout
[crazyswarm_server-2] process has died [pid 5565, exit code -6, cmd /home/santiagorg2401/crazyflie/crazyswarm/ros_ws/devel/lib/crazyswarm/crazyswarm_server __name:=crazyswarm_server __log:=/home/santiagorg2401/.ros/log/3456f9ce-cdbb-11ea-ab55-181deab367da/crazyswarm_server-2.log].
log file: /home/santiagorg2401/.ros/log/3456f9ce-cdbb-11ea-ab55-181deab367da/crazyswarm_server-2*.log
`
Terminal 3
`santiagorg2401@santiagorg2401-G7-7588:~$ source crazyflie/crazyswarm/ros_ws/src/crazyswarm/scripts/niceHover.py 
import-im6.q16: not authorized `np' @ error/constitute.c/WriteImage/1037.
from: can't read /var/mail/pycrazyswarm
Z: orden no encontrada
bash: crazyflie/crazyswarm/ros_ws/src/crazyswarm/scripts/niceHover.py: línea 9: error sintáctico cerca del elemento inesperado `('
bash: crazyflie/crazyswarm/ros_ws/src/crazyswarm/scripts/niceHover.py: línea 9: `    swarm = Crazyswarm()'
santiagorg2401@santiagorg2401-G7-7588:~$ source crazyflie/crazyswarm/ros_ws/devel/setup.bash 
santiagorg2401@santiagorg2401-G7-7588:~$ python crazyflie/crazyswarm/ros_ws/src/crazyswarm/scripts/niceHover.py 
Traceback (most recent call last):
  File "crazyflie/crazyswarm/ros_ws/src/crazyswarm/scripts/niceHover.py", line 9, in <module>
    swarm = Crazyswarm()
  File "/home/santiagorg2401/crazyflie/crazyswarm/ros_ws/src/crazyswarm/scripts/pycrazyswarm/crazyswarm.py", line 36, in __init__
    crazyflies_yaml = open(crazyflies_yaml, 'r').read()
IOError: [Errno 2] No such file or directory: '../launch/crazyflies.yaml'
`

**Comment by whoenig** [comment]
- The timeout error indicates that you can't connect to the CF. Check your URI `radio://0/80/2M/E7E7E7E706` and that you can connect to that CF with other means, such as the `cfclient`.
- The niceHover.py script needs to be launched from the scripts folder, not the home-folder, otherwise the relative paths don't work.

**Comment by santiagorg2401** [comment]
I will try again when being in the lab and I will post the results.
My brother is covid-19 suspect, so I have to wait for his results.
Thank you for answering Mr. Whönig

**Comment by santiagorg2401** [comment]
Hi, I re-installed all the packages, and this is what happens:
![image](https://user-images.githubusercontent.com/52050852/89429672-7a014a00-d703-11ea-8820-bbd556b17359.png)
Checking the address witch cfclient:
![image](https://user-images.githubusercontent.com/52050852/89429798-a0bf8080-d703-11ea-9301-c2f0a41151ec.png)

**Comment by santiagorg2401** [comment]
Update:
The battery button is still not working but I could make them fly, but there are 2 problems, sometimes Rviz does not connect to the Crazyflie and it takes several tries until it connects, and many times when the Crazyflie starts flying, it just goes anywhere out the trajectory and crashes

**Comment by whoenig** [comment]
- For the battery, make sure you use a recent NRF51 firmware and crazyswarm. In particular, make sure you picked up [this fix](https://github.com/bitcraze/crazyflie2-nrf-firmware/commit/eaffde62d30358668647ac05bacc6a02f2241d1c).
- For the random fly-aways, make sure that the state estimation is correct. The typical approach is to enable logging, log `stateEstimate.x`, `stateEstimate.y`, `stateEstimate.z`, run `hover_swarm.launch` (but do not attempt to fly), and move the Crazyflie manually in the space. If you use `rosecho /cf8/log1`, you should see the correct coordinates. Also keep in mind that there seem to be problems with the state estimator, see [this issue](https://github.com/bitcraze/crazyflie-firmware/issues/555).
- For the connection issue, please state what the error is when there is a failure. Perhaps you are running into another [firmware bug](https://github.com/USC-ACTLab/crazyswarm/issues/182)? 

Also, I hope both you and your brother are fine and Covid-free!

**Comment by santiagorg2401** [comment]
Hi,
For the firmware updating I am using the cfclient GUI app, is that ok?
For the estimators, do I have to type them on the Python code in crazyswarm/ros_ws/src/crazyswarm/scripts ?
I will attach a screenshot (of the Rviz issue) when being in the laboratory.
Thank you for your hopes, and yes, we are covid free.
